### ISSUE TITLE
Error on ray.init when using --redis-password

### ISSUE THREAD

**Issue by dhroth** [ray-project/ray#3082]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04 (Linux c70 4.15.0-29-generic #31~16.04.1-Ubuntu SMP Wed Jul 18 08:54:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux)  
- **Ray installed from (source or binary)**: `pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.5.3-cp35-cp35m-manylinux1_x86_64.whl`
- **Ray version**: 0.5.3
- **Python version**:  3.5.2
- **Exact command to reproduce**:
```
(tflow) [drothchild@c70:~/sgd/subspace]$ export RAY_USE_XRAY=1
(tflow) [drothchild@c70:~/sgd/subspace]$ ray start --head --redis-port=34959 --num-gpus=1 --redis-password="abcdefgHIJKLMNOP"
Detected environment variable 'RAY_USE_XRAY'.
Using IP address 169.229.49.170 for this node.
Process STDOUT and STDERR is being redirected to /tmp/ray/session_2018-10-17_22-57-05_5704/logs.
Waiting for redis server at 127.0.0.1:34959 to respond...
Waiting for redis server at 127.0.0.1:27888 to respond...
Warning: Reducing object store memory because /dev/shm has only 32110059520 bytes available. You may be able to free up space by deleting files in /dev/shm. If you are inside a Docker container, you may need to pass an argument with the flag '--shm-size' to 'docker run'.
Starting the Plasma object store with 25.69 GB memory.
Failed to start the UI, you may need to run 'pip install jupyter'.
{'object_store_addresses': [ObjectStoreAddress(name='/tmp/ray/session_2018-10-17_22-57-05_5704/sockets/plasma_store', manager_name=None, manager_port=None)], 'raylet_socket_names': ['/tmp/ray/session_2018-10-17_22-57-05_5704/sockets/raylet'], 'node_ip_address': '169.229.49.170', 'local_scheduler_socket_names': [], 'webui_url': None, 'redis_address': '169.229.49.170:34959'}

Started Ray on this node. You can add additional nodes to the cluster by calling

    ray start --redis-address 169.229.49.170:34959 --redis-password abcdefgHIJKLMNOP

from the node you wish to add. You can connect a driver to the cluster from Python by running

    import ray
    ray.init(redis_address="169.229.49.170:34959", redis_password="abcdefgHIJKLMNOP")

If you have trouble connecting from a different machine, check that your firewall is configured properly. If you wish to terminate the processes that have been started, run

    ray stop
(tflow) [drothchild@c70:~/sgd/subspace]$ python
Python 3.5.2 (default, Nov 23 2017, 16:37:01)
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ray
>>> ray.init(redis_address="169.229.49.170:34959", redis_password="abcdefgHIJKLMNOP")
Detected environment variable 'RAY_USE_XRAY'.
Some processes that the driver needs to connect to have not registered with Redis, so retrying. Have you run 'ray start' on this node?
Some processes that the driver needs to connect to have not registered with Redis, so retrying. Have you run 'ray start' on this node?
Some processes that the driver needs to connect to have not registered with Redis, so retrying. Have you run 'ray start' on this node?
Some processes that the driver needs to connect to have not registered with Redis, so retrying. Have you run 'ray start' on this node?
Some processes that the driver needs to connect to have not registered with Redis, so retrying. Have you run 'ray start' on this node?
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/work/drothchild/virtualenvs/tflow/lib/python3.5/site-packages/ray/worker.py", line 1804, in init
    temp_dir=temp_dir)
  File "/work/drothchild/virtualenvs/tflow/lib/python3.5/site-packages/ray/worker.py", line 1611, in _init
    redis_password=redis_password)
  File "/work/drothchild/virtualenvs/tflow/lib/python3.5/site-packages/ray/worker.py", line 1345, in get_address_info_from_redis
    redis_password=None)
  File "/work/drothchild/virtualenvs/tflow/lib/python3.5/site-packages/ray/worker.py", line 1297, in get_address_info_from_redis_helper
    clients = redis_client.zrange(client_key, 0, -1)
  File "/work/drothchild/virtualenvs/tflow/lib/python3.5/site-packages/redis/client.py", line 1754, in zrange
    return self.execute_command(*pieces, **options)
  File "/work/drothchild/virtualenvs/tflow/lib/python3.5/site-packages/redis/client.py", line 668, in execute_command
    return self.parse_response(connection, command_name, **options)
  File "/work/drothchild/virtualenvs/tflow/lib/python3.5/site-packages/redis/client.py", line 680, in parse_response
    response = connection.read_response()
  File "/work/drothchild/virtualenvs/tflow/lib/python3.5/site-packages/redis/connection.py", line 629, in read_response
    raise response
redis.exceptions.ResponseError: NOAUTH Authentication required.
>>> 
```

### Describe the problem
The `init()` call works as expected if I omit the `--redis-password` arg from `ray start` and remove the `redis_password` python kwarg.

### Source code / logs
Full traceback above.

**Comment by dhroth** [comment]
Thanks. I tried commenting out that line but still get the error. What do you mean by running that `redis-cli` command for both redis servers? I could run that command after calling `ray start`, but the `ray.init()` function throws an error, so I'm not sure how I'd run `redis-cli` on the relevant port. (Also, I'm running this on a Millennium machine that doesn't have `redis-cli`)

**Comment by pschafhalter** [comment]
I think it might be a typo when passing the `redis_password` parameter (see #3083). Sorry about that.

**Comment by robertnishihara** [comment]
Fixed by #3083
